sdks/python/apache_beam/utils/annotations_test.py

```diff
@@ -0,0 +1,101 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import unittest
+import warnings
+from annotations import deprecated
+from annotations import experimental
+
+
+class AnnotationTests(unittest.TestCase):
+  # Note: use different names for each of the the functions decorated
+  # so that a warning is produced for each of them.
+  def test_deprecated_with_since_current(self):
+    with warnings.catch_warnings(record=True) as w:
+      @deprecated(since='v.1', current='multiply')
+      def fnc_test_deprecated_with_since_current():
+        return 'lol'
+      fnc_test_deprecated_with_since_current()
+      self.assertEqual(1, len(w))
```

Comment:
My thinking was to have each code read easily as a unit, but I anyway added the check_annotation method with the common parts of the code.